src/style/font/ttf.rs

```diff
@@ -159,31 +159,25 @@ impl FontData for FontDataInternal {
 
     fn draw<E, DrawFunc: FnMut(i32, i32, f32) -> Result<(), E>>(
         &self,
-        (x, y): (i32, i32),
+        (base_x, base_y): (i32, i32),
         size: f64,
         text: &str,
-        trans: FontTransform,
         mut draw: DrawFunc,
     ) -> Result<Result<(), E>, Self::ErrorType> {
-        let layout = self.estimate_layout(size, text)?;
-
         let scale = Scale::uniform(size as f32);
         let mut result = Ok(());
         let font = &self.0;
 
-        let base_x = x + trans.offset(layout).0;
-        let base_y = y + trans.offset(layout).1;
-
         for g in font.layout(text, scale, point(0.0, 0.0)) {
             if let Some(rect) = g.pixel_bounding_box() {
-                let x0 = rect.min.x;
```

Comment:
And I believe this is one of the reason why the descender is out of the box now.

Reply:
Yes, it's a regression bug I've introduced, the up-to-dated change has the proper calculations.